app/src/main/java/ch/onepass/onepass/model/user/UserRepositoryFirebase.kt

```
@@ -99,6 +102,37 @@ class UserRepositoryFirebase(
     }
   }
 
+  override fun getFavoriteEvents(uid: String): Flow<Set<String>> = callbackFlow {
+    val docRef = userCollection.document(uid)
+    val listener =
+        docRef.addSnapshotListener { snapshot, error ->
+          if (error != null) {
+            close(error)
+            return@addSnapshotListener
+          }
+          if (snapshot != null && snapshot.exists()) {
+            val user = snapshot.toObject(User::class.java)
+            val favorites = user?.favoriteEventIds?.toSet() ?: emptySet()
+            trySend(favorites)
+          } else {
+            trySend(emptySet())
+          }
+        }
+    awaitClose { listener.remove() }
+  }
+
+  override suspend fun addFavoriteEvent(uid: String, eventId: String): Result<Unit> = runCatching {
+    userCollection.document(uid).update("favoriteEventIds", FieldValue.arrayUnion(eventId)).await()
+  }
+
+  override suspend fun removeFavoriteEvent(uid: String, eventId: String): Result<Unit> =
+      runCatching {
+        userCollection
+            .document(uid)
+            .update("favoriteEventIds", FieldValue.arrayRemove(eventId))
```

Good. Silently succeeds if eventId doesn't exist (Firestore behavior).